Sdk/MoyasarSdk/Views/STCPayModule/View/STCPayView.swift

```diff
@@ -0,0 +1,54 @@
+//
+//  STCPayView.swift
+//  MoyasarSdk
+//
+//  Created by Mahmoud Abdelwahab on 11/09/2024.
+//
+
+import SwiftUI
+
+public struct STCPayView: View {
+    /// The observed view model that handles the STC Pay logic.
+    @ObservedObject var viewModel: STCPayViewModel
+    
+    /// Initializes the `STCPayView` with a payment request and a callback for handling the payment result.
+      ///
+      /// - Parameters:
+      ///   - paymentRequest: The request containing payment details, including the API key.
+      ///   - callback: A closure that gets called when the payment process finishes with success or failure.
+      ///
+      /// - Throws: If the `STCPayViewModel` initialization fails due to an invalid API key,
+      /// it throws an error, which is handled by showing a fatal error message.
+    public init(paymentRequest: PaymentRequest, callback: @escaping STCResultCallback) {
+        do {
+            viewModel = try STCPayViewModel(paymentRequest: paymentRequest, resultCallback: callback)
+        } catch {
+            // Handle error here, show error in view model
+            fatalError("Invalid api key 🙁")
```

Comment:
I'm not that familiar with Swift, but it seems this catch is a catch-all block, yet it only shows an error related to the API Key. 

It may be confusing if any error not related to the API key is suppressed here with with a message related to the API Key. 

What do you think?

Moreover, are the errors, listed in the following link, will be caught here? If not, how are they handled?

https://www.notion.so/Mobile-SDK-Feature-Guideline-0538ee36313046d690f82a3e02077409?pvs=4#62fb17927925477eac252c3c2ca18c01

Reply:
- Yes i already handle all theses errors in the  resultCallback it has success with ApiPayment response and failure with moyasar error return the descriptive message from backend 
-  This catch for api key validation error  i ensure here that if client entered wrong api key for live or test to not allow him to continue create payment service at all this show api key error in compile time to avoid customers from getting that error later when send request or something else

Reply:
Great that you already handle listed errors 🚀

> This catch for api key validation error

Please forgive my limited experience with Swift, but I don't see how errors, other than API Key validity, won't be suppressed here by this message. I hope you see where I'm coming from. 🤔

Reply:
The constructor has payment request and STCResultCallback this STCResultCallback provide all outputs for user like all listed errors  
init(paymentRequest: PaymentRequest, callback: @escaping STCResultCallback)